### ISSUE TITLE
Unable to download Java JDK/SRE archives during build

### ISSUE THREAD

**Issue by Carmageddon** [anapsix/docker-alpine-java#82]
The command:
curl -jksSLH 'Cookie: oraclelicense=accept-securebackup-cookie' -o /tmp/java.tar.gz http://download.oracle.com/otn-pub/java/jdk/8u202-b08/1961070e4c9b4e26a04e7f5a083f551e/jdk-8u202-linux-x64.tar.gz
is downloading HTML content titled "Error Page 404" from Oracle

I am not sure why, has Oracle stopped serving JDK8?

**Comment by anapsix** [comment]
ugh! other than, new versions being available, looks like Oracle now redirects to sign-in page
try it yourself from [JDK8 Download Page](https://www.oracle.com/technetwork/pt/java/javase/downloads/jdk8-downloads-2133151.html)
Will have to figure out how to work it.. 
For now, just use pre-built images from Docker Hub

**Comment by anapsix** [comment]
hmm also
```
Important Oracle JDK License Update

The Oracle JDK License has changed for releases starting April 16, 2019.

The new Oracle Technology Network License Agreement for Oracle Java SE
is substantially different from prior Oracle JDK licenses. The new license permits
certain uses, such as personal use and development use, at no cost -- but other
uses authorized under prior Oracle JDK licenses may no longer be available. Please
review the terms carefully before downloading and using this product. An FAQ is
available here. 

Commercial license and support is available with a low cost Java SE Subscription. 

Oracle also provides the latest OpenJDK release under the open source GPL License at jdk.java.net.
```

**Comment by llamahunter** [comment]
time to switch to openjdk?

**Comment by anapsix** [comment]
perhaps it is @llamahunter..
or alternatively, if one needs Oracle Java, one can sign up for an account, download files locally, and modify relevant Dockerfile to do `COPY ./java.tar.gz /tmp/java.tar.gz`. After all minimization steps are completed, using another stage, one should copy the content of `/opt/jdk/` into prepared in another stage Alpine image with Glibc..

This worries me
```
The new license permits certain uses, such as personal use
and development use, at no cost -- but other uses authorized
under prior Oracle JDK licenses may no longer be available.
```
```

**Comment by anapsix** [comment]
Right.. as announced, Java licensing changed, and starting April 2019 commercial usage of Oracle Java required subscription. In other words, switch to OpenJDK, or use older versions of Oracle Java (for reference, pre-built images are available on [Docker Hub](https://hub.docker.com/r/anapsix/alpine-java/))

Official OpenJDK images receive regular updates, and are available at https://hub.docker.com/_/openjdk, including `8-jre-alpine` (85MB), `8-jdk-alpine` (105MB), `8-jre-slim` (204MB), `8-jdk-slim` (243MB), `8u212-jre-slim` (204MB), `8u212-jdk-slim` (243MB), etc..

For more details about Oracle Java Licensing, checkout this article: https://medium.com/@javachampions/java-is-still-free-2-0-0-6b9aa8d6d244
TL;DR
> Oracle JDK 8 is going through the “End of Public Updates” process, which means the April 2019 update will restrict commercial use. However, since Java SE 9, Oracle is also providing Oracle OpenJDK builds which are free for commercial use (but only updated for 6 months). There are also free OpenJDK builds which will be updated (including security patches) from other providers like AdoptOpenJDK, Amazon, Azul, BellSoft, IBM, jClarity, Red Hat, the Linux distros et al.

And many others talking about Oracle Java SE licensing changes ([Google search](https://www.google.com/search?q=java+oracle+license))
